src/rust/ide/ast/impl/src/lib.rs

```diff
@@ -272,11 +271,6 @@ impl Ast {
     pub fn with_shape<S:Into<Shape<Ast>>>(&self, shape:S) -> Ast {
         Ast::new(shape.into(),self.id)
     }
-
-    /// Returns this AST node with removed ID.
-    pub fn without_id(&self) -> Ast {
-        Ast::from_ast_id_len(self.shape().clone(), None, self.len())
-    }
```

Comment:
I'd keep this as long as the Ast represents Id as an Option.  Particularly if we allow calling `from_ast_id_len` with `None`.
If this allows breaking our rules (assuming we adopt stricter take on them), make this private.